src/initialization/simpleinit.jl

```diff
@@ -0,0 +1,34 @@
+function initial_update!(integ, cache, init::SimpleInit)
+    @unpack u, f, p, t = integ
+    @unpack du, x, Proj = cache
+
+    if f isa ODEFunction &&
+       f.f isa SciMLBase.FunctionWrappersWrappers.FunctionWrappersWrapper
+        f = ODEFunction(SciMLBase.unwrapped_f(f), mass_matrix=f.mass_matrix)
+    end
+
+    # This is hacky and should definitely be removed. But it also works so 🤷
+    MM = if f.mass_matrix isa UniformScaling
```

Comment:
Move this to a separate function as right now this is duplicated.

Reply:
Thinking about it, maybe I should merge the simple init and the autodiff init? In principle that should work and provide the cleanest code. The only downside is probably the additional allocaton of the latter, but if I pre-allocate the memory for where to store the derivatives in then it should be fine right?